cmd/maya-exporter/app/collector/collector.go

```diff
@@ -122,6 +137,17 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 		stats       stats
 	)
 
+	c.Lock()
+	if c.isRequestInProgress() {
+		c.targetRejectRequestCounter.Inc()
+		c.targetRejectRequestCounter.Collect(ch)
+		c.Unlock()
```

Comment:
Related to my above comment on pushing lock/unlock to RequestInProgress, will this sequence work?

```
		c.Unlock()
		c.targetRejectRequestCounter.Inc()
		c.targetRejectRequestCounter.Collect(ch)
```

Reply:
channel itself is protected by lock so sequence might change because after unlock we don't know which request came first and which one incremented the the request counter. Though it doesn't matter as we are only incrementing counters, so above sequence will work also.

Reply:
yes